modules/overriding-default-node-ip-selection-logic.adoc

```diff
@@ -0,0 +1,142 @@
+// This is included in the following assemblies:
+//
+// * installing/installing_bare_metal_ipi/ipi-install-installation-workflow.adoc
+
+:_content-type: PROCEDURE
+[id="overriding-default-node-ip-selection-logic_{context}"]
+= Optional: Overriding the default node IP selection logic 
+
+In most circumstances, the `NODEIP_HINT` is the API VIP. The IP selection logic will determine which subnet the `NODEIP_HINT` belongs to, and from that determine the interface in that subnet and select an IP on that interface. For example, a node that has two interfaces, `eth0` with the subnet range `10.0.0.10/24`, and `eth1` with the subnet range `192.0.2.5/24`, and with a default route via the `10.0.0.10` source address on `eth0`:
```

Comment:
"In most circumstances, the `NODEIP_HINT` is the API VIP."

Something is not right here. This procedure doesn't apply to IPI platforms where the VIP is used for ip selection. It can only be used on UPI deployments that don't have VIPs at all. I see on line 3 that this is being included in only the IPI docs when it should be included in only the UPI docs.

Reply:
@cybertron I apologize for the confusion, but I thought this content was supposed to go here, based on your comment in Slack? See: 

This could go in the troubleshooting network issues section. We'd actually prefer people use this new method than the one already in the docs, although that will continue to work too for the rare situation where the hint file isn't sufficient. The other place this could go is under the manifests section of the installation docs: [https://docs.openshift.com/container-platform/4.11/installing/installing_bare_metal_ipi/ipi-install-installation-workflow.html#ipi-insta[…]ion-files](https://docs.openshift.com/container-platform/4.11/installing/installing_bare_metal_ipi/ipi-install-installation-workflow.html#ipi-install-manifest-configuration-files) The mechanism for this is similar to the other things we're already doing there.

The manifest section falls under the ipi-install-installation-workflow assembly, which is this page: https://52200--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal_ipi/ipi-install-installation-workflow.html#installing-rhel-on-the-provisioner-node_ipi-install-installation-workflow

Reply:
Oops, yeah that was my mistake. I should not have suggested that. Maybe it's easiest to leave it in the troubleshooting section? It probably could go in the similar UPI section: https://docs.openshift.com/container-platform/4.11/installing/installing_bare_metal/installing-bare-metal-network-customizations.html#modifying-nwoperator-config-startup_installing-bare-metal-network-customizations

But since this applies to other on-prem platforms besides baremetal maybe the generic troubleshooting section is best.